File: C++/freeCodeCamp.org_C++ Tutorial for Beginners/src/026_Object Functions/ObjectFunctions.cpp
#include <iostream>

/* Simple C++ program that uses object functions */

using namespace std;

class Student
{
    public:
        string name;
        string major;
        double gpa;
        Student(string argName, string argMajor, double argGPA)
        {
            cout << "[DEBUG]: Creating object [\"" << argName << "\",\"" << argName << "\", \"" << argName << "\"]..." << endl;
            name = argName;
            major = argMajor;
            gpa = argGPA;
        }

        bool hasHonors()
        {
            if(gpa >= 3.5)
            {
                return true;
            } else {
                return false;
            }
        }

};

int main()
{
    Student student1("Jim", "Business", 2.4);
    Student student2("Pam", "Art", 3.6);
    Student student3("Tom", "Art", 4.0);
    Student student4("Sam", "Art", 5.0);
    
    cout << student1.hasHonors() << endl;
    cout << student2.hasHonors() << endl;
    cout << student3.hasHonors() << endl;

    return 0;
}

File: Python/002_OOP Basics/026_Object Functions/ObjectFunctions.py
class Student:

    name = ""
    major = ""
    gpa = float(0)

    def __init__(self, name="No name", major="No major", gpa=0.0):
        print("\n[DEBUG]: Creating student object [\"", name, "\", \"", major, "\", \"", gpa, "\"]...")
        self.name = name
        self.major = major
        self.gpa = gpa
        print("[DEBUG]: Done")

    def hasHonors(self):
        if(self.gpa >= 3.5):
            return True
        else:
            return False

student1 = Student("Jim", "Business", 2.4)
student2 = Student("Pam", "Art", 3.6)
student3 = Student("Tom", "Art", 4.0)
student4 = Student("Sam", "Art", 5.0)

print("\nHas honors:\t", student1.hasHonors())
print("Has honors:\t", student2.hasHonors())
print("Has honors:\t", student3.hasHonors())
print("Has honors:\t", student4.hasHonors())